sql/connect/common/src/main/scala/org/apache/spark/sql/connect/client/SparkConnectClient.scala

```diff
@@ -423,7 +423,6 @@ object SparkConnectClient {
     def configuration: Configuration = _configuration
 
     def userId(id: String): Builder = {
-      // TODO this is not an optional field!
```

Comment:
We might also need to fix `SparkConnectClient.__init__` here:

```
            self._user_id = os.getenv("USER", None)
```

to

```
            self._user_id = os.getenv("SPARK_USER", os.getenv("USER", None))
```

Reply:
Thank you for review, @HyukjinKwon . I'll take a look at that part too.